scripts/veloxbench/veloxbench/_benchmark.py

```diff
@@ -0,0 +1,238 @@
+    ):
+        if name is None:
+            name = self.name
+        tags, info, context = self._get_tags_info_context(case, extra_tags)
+        info.update(**extra_info)
+        context.update(**extra_context)
+
+        benchmark, output = self.conbench.record(
+            result,
+            name,
+            tags=tags,
+            info=info,
+            context=context,
+            github=self.github_info,
+            options=options,
+            output=output,
+            publish=os.environ.get("DRY_RUN") is None,
+        )
+        return benchmark, output
+
+    def execute_command(self, command):
+        try:
+            print(command)
+            result = subprocess.run(command, check=True)
+        except subprocess.CalledProcessError as e:
+            print(e.stderr.decode("utf-8"))
+            raise e
+
+    def version_case(self, case: tuple) -> Optional[int]:
+        """
+        A method that takes a case and returns a version int
+
+        :param case: A tuple with a value for each argument of the benchmark's parameters
+
+        Overwrite this method to version cases. Incrementing versions will break case history
+        because `case_version` will be appended to tags, which are considered in determining
+        history. Returning `None` corresponds to no versioning.
+        """
+        return None
+
+    def get_tags(
+        self, options: Dict[str, Any], source: Optional[list] = None
+    ) -> Dict[str, Any]:
+        cpu_tag = {"cpu_count": options.get("cpu_count", None)}
+        if source:
+            return {**source.tags, **cpu_tag}
+        else:
+            return cpu_tag
+
+    def _get_tags_info_context(
+        self, case: tuple, extra_tags: Dict[str, Any]
+    ) -> Tuple[Dict[str, Any], Dict[str, Any], Dict[str, Any]]:
+        info = {**self.velox_info}
+        info.pop("velox_git_revision")
+
+        context = {"velox_compiler_flags": info.pop("velox_compiler_flags")}
+
+        tags = {**extra_tags}
+
+        if case:
+            case_tags = dict(zip(self.fields, case))
+            for key in case_tags:
+                if key not in tags:
+                    tags[key] = case_tags[key]
+
+            case_version = self.version_case(case)
+            if case_version:
+                tags["case_version"] = case_version
+
+        return tags, info, context
+
+    def _handle_error(
+        self,
+        e: Exception,
+        name: str,
+        tags: Dict[str, Any],
+        info: Dict[str, Any],
+        context: Dict[str, Any],
+    ) -> Tuple[Dict[str, Any], None]:
+        output = None
+        tags["name"] = name
+        error = {
+            "timestamp": _now_formatted(),
+            "tags": tags,
+            "info": info,
+            "context": context,
+            "error": str(e),
+        }
+        logging.exception(json.dumps(error))
+        return error, output
+
+
+@conbench.runner.register_list
+class BenchmarkList(conbench.runner.BenchmarkList):
```

Comment:
Do we really need this?

Will anybody/anything use `conbench list`?

Reply:
Yeah, this is probably a bit over-kill here. We are planning to revisit if we want to unify the current benchmark runner script and we can take this out if that runner script will be the entrypoint for developers locally. 

For what it's worth, I found the list to be pretty helpful in testing locally that veloxbench was in the right place.